lib/ruby_indexer/lib/ruby_indexer/reference_finder.rb

```diff
@@ -264,13 +264,51 @@ def on_def_node_leave(node)
 
     sig { params(node: Prism::CallNode).void }
     def on_call_node_enter(node)
-      if @target.is_a?(MethodTarget) && (name = node.name.to_s) == @target.method_name
+      return unless @target.is_a?(MethodTarget)
+
+      if (name = node.name.to_s) == @target.method_name
         @references << Reference.new(name, T.must(node.message_loc), declaration: false)
+      elsif attr_method_references?(node)
+        @references << Reference.new(@target.method_name, T.must(node.message_loc), declaration: true)
       end
     end
 
     private
 
+    sig { params(node: Prism::CallNode).returns(T::Boolean) }
+    def attr_method_references?(node)
+      case node.name
+      when :attr_reader
+        attr_reader_references?(unescaped_argument_names(node))
+      when :attr_writer
+        attr_writer_references?(unescaped_argument_names(node))
+      when :attr_accessor
+        attr_accessor_references?(unescaped_argument_names(node))
+      else
+        false
+      end
+    end
+
+    sig { params(node: Prism::CallNode).returns(T::Array[String]) }
+    def unescaped_argument_names(node)
+      node.arguments.arguments.select { |arg| arg.respond_to?(:unescaped) }.map(&:unescaped)
```

Comment:
In the middle of typing, we may find empty arguments, so we have to handle that. Also, we need to account for symbol and string nodes as arguments.

```suggestion
      arguments = node.arguments.arguments
      return [] unless arguments

      arguments.filter_map do |arg|
        case arg
        when Prism::StringNode
          arg.unescaped
        when Prism::SymbolNode
          arg.value
        end
      end
```